FireboltNETSDK/FireboltClient.cs

```diff
@@ -226,6 +238,44 @@ protected async Task<T> SendAsync<T>(HttpMethod method, string uri, HttpContent?
 
         try
         {
+            string updateParametersHeader = "Firebolt-Update-Parameters";
+            if (response.Headers.Contains(updateParametersHeader))
+            {
+                string? newDatabase = null;
+                string? newEngine = null;
+                bool reconnect = false;
+                foreach (string[] kv in response.Headers.GetValues(updateParametersHeader).Select(p => p.Split('=', 2, StringSplitOptions.TrimEntries)))
+                {
+                    if (kv.Length == 1 || "".Equals(kv[1]))
+                    {
+                        connectionParameters.Remove(kv[0]);
+                    }
+                    else
+                    {
+                        connectionParameters[kv[0]] = kv[1];
+                        switch (kv[0])
+                        {
+                            case "database":
+                                newDatabase = kv[1];
+                                reconnect = true;
+                                break;
+                            case "engine":
+                                newEngine = kv[1];
+                                reconnect = true;
+                                break;
+                        }
+                    }
+                }
+                if (reconnect)
+                {
+                    if (newEngine != null)
+                    {
+                        _connection.EngineName = newEngine;
+                    }
+                    _connection.ChangeDatabase(newDatabase ?? _connection.Database);
+                }
```

Comment:
Why do we need to reconnect?

Reply:
Unfortunately yes because we have to change the state (in this case DB name). This is exactly what ChangeDatabase() does. The flag "reconnect" is probably a little bit confusing but in fact it just means that `ChangeDatabase()` should be called that in turn means that either DB or engine has been changed.